core/xim/pm/XPackage.lua

```diff
@@ -40,33 +40,49 @@ function XPackage:name()
     return self.pdata.name
 end
 
-function XPackage:support()
-    local pm = self.pdata.pmanager
-    if pm[self.version][os_info.name] then
-        return true
+function XPackage:xpm_enable()
+    local xpm = self.pdata.xpm
+    if not xpm or not xpm[os_info.name] then
+        return false
     end
-    return false
+    return xpm[os_info.name][self.version] ~= nil
 end
 
-function XPackage:get_xpm()
-    local pms = self:get_pmanager()
-    return pms.xpm
+function XPackage:get_xpm_resources()
+    return self.pdata.xpm[os_info.name][self.version]
 end
 
 function XPackage:deps()
-    if not self.pdata.deps then
+    if not self.pdata.xpm or not self.pdata.xpm[os_info.name] then
         return {}
     end
-    return self.pdata.deps[os_type]
+    return self.pdata.xpm[os_info.name].deps or {}
 end
 
-function XPackage:get_pmanager()
-    local pm = self.pdata.pmanager
-    if not pm then
-        cprint("[xlings:xim]: get_pmanager: package manager not found")
-        return { xpm = { url = nil, sha256 = nil } }
+function XPackage:get_pm_wrapper()
+    -- self.version is package manager name when use local package manager
+    return self.version
+end
+
+function XPackage:get_map_pkgname()
+    return self.pdata.pm_wrapper[self.version]
+end
```

Comment:
It's not well-defined here. Either a design slip or a impl slip.

Reply:
`XPackage` is runtime object, so if you just do that, why we just read the file and extract essential thing🤔? The runtime initiation isn't necessary. And `Service` itself just a middleware, there's no need to create an instance for doing that. The package index should be what you said in `pkgindex`, `XPackage` could be an initiation of a certain sibling, to maintain the independence.